prow/plugins/project/project.go

```diff
@@ -0,0 +1,348 @@
+		if org == orgName {
+			return orgConfig.ProjectColumnMap
+		}
+	}
+	return nil
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	configInfo := map[string]string{}
+	for _, repo := range enabledRepos {
+		parts := strings.Split(repo, "/")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid repo in enabledRepos: %q", repo)
+		}
+		if maintainerTeamID, maintainerTeamName := getMaintainerTeam(config.Project, parts[0], parts[1]); maintainerTeamID != 0 && maintainerTeamName != "" {
+			configInfo[repo] = fmt.Sprintf(projectTeamMsg, maintainerTeamName, maintainerTeamID)
+		} else {
+			configInfo[repo] = "There are no maintainer team specified for this repo or its org."
+		}
+
+		if columnMap := getColumnMap(config.Project, parts[0], parts[1]); len(columnMap) != 0 {
+			configInfo[repo] = fmt.Sprintf(columnsMsg, columnMap)
+		} else {
+			configInfo[repo] = "There are not default columns specified for this repo."
+		}
+	}
+
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The project plugin allows members of a GitHub team to set the project and column on an issue or pull request.",
+		Config:      configInfo,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/project <board>, /project <board> <column>, or /project clear <board>",
+		Description: "Add an issue or PR to a project board and column",
+		Featured:    false,
+		WhoCanUse:   "Members of the project maintainer GitHub team can use the '/project' command.",
+		Examples:    []string{"/project 0.5.0", "/project 0.5.0 To do", "/project clear 0.4.0"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.PluginConfig.Project)
+}
+
+func updateProjectNameToIDMap(projects []github.Project) {
+	for _, project := range projects {
+		projectNameToIDMap[project.Name] = project.ID
+	}
+}
+
+func parseCommand(command string) (string, string, bool, string) {
+	var shouldClear = false
+	matches := projectRegex.FindStringSubmatch(command)
+
+	// No project is provided
+	if len(matches) == 0 {
+		msg := invalidNumArgs
+		return "", "", false, msg
+	}
+	proposedProject := matches[1]
+	proposedColumnName := ""
+	if len(matches) > 1 && proposedProject != clearKeyword {
+		proposedColumnName = matches[2]
+	}
+	// If command is to clear and the project is provided
+	if proposedProject == clearKeyword {
+		if len(matches) > 2 && matches[2] != "" {
+			proposedProject = matches[2]
+			shouldClear = true
+		} else {
+			proposedProject = ""
+		}
+	}
+
+	return proposedProject, proposedColumnName, shouldClear, ""
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfig plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	proposedProject := ""
```

Comment:
The construct

```go
if foo := doSomething(); foo == "something" {
```

construct is useful since it allows for `foo` to be scoped to the block where it's used, but here you're assigning to `proposedProject` _etc_ regardless, so instead you could just call `parseCommand()` at the top level and not need these default values